### PR TITLE
Angular points

### PR DESCRIPTION
**Pull request by edgargdev**
I need to do some refactoring of the authentication middleware. I just copied it over to the venues route file just so I could build out the backend points feature.

### CODE REVIEW

File: app/public/scripts/angular/panelStates/visited.js [hunk 1]
```diff
@@ -31,8 +35,8 @@ function VisitedController() {
       });
       bar.text.style.fontFamily = '"Raleway", Helvetica, sans-serif';
       bar.text.style.fontSize = '2rem';
-
-      bar.animate(0.5);  // Number from 0.0 to 1.0
+      let percentage = user.points / venues.length
```
Comment (michaelachrisco) [inline app/public/scripts/angular/panelStates/visited.js:38]: Nice!

File: app/backend/routes/venues.js [hunk 2]
```diff
@@ -10,4 +13,59 @@ router.get('/', (req,res,next) => {
...
+      })
+    })
+  } else {
+    res.send({text: "Sorry key doesn't match."})
+  }
+})
+
 module.exports = router
+
+function ensureAuthenticated(req, res, next) {
```
Comment (michaelachrisco) [inline app/backend/routes/venues.js]: I would add this check into the users model in `../models/user.js` or as a `../services/user.js` and use the method as a service.

File: app/backend/routes/venues.js [hunk 3]
```diff
@@ -10,4 +13,59 @@ router.get('/', (req,res,next) => {
   res.send(output)
 })
 
+router.post('/verify',ensureAuthenticated, (req, res, next) => {
```
Comment (michaelachrisco) [inline app/backend/routes/venues.js]: At this point, do you mind refactoring this method into own controller and documented. This route file is doing a bit too much and is going to cause future confusion. I believe you alluded a bit to this.

**Review by michaelachrisco — CHANGES_REQUESTED** [review]
Good PR. Just push the user route methods into the user model or a middleware/service and I think we should be good.

**Comment by michaelachrisco** [comment]
As always @rxsharp  can override my opinion and merge, but I believe a small refactor needs to be done before this branch is merged. Not a huge deal but the route files are doing things the angular controllers (or middlewere depending) should be doing.

Thanks for the PR @Rawradin! Just make those small changes and Ill merge.

**Comment by edgargdev** [comment]
Refactored most of the addressed points but could use a second opinion on how to proceed with all the User updating points logic in /venues/verify route.

**Comment by rick** [comment]
[motor away](https://www.youtube.com/watch?v=9J-V6AGuA2k)

**Comment by michaelachrisco** [comment]
@Rawradin LGTM!

**Pull request merged**